src/main/java/uk/gov/hmcts/reform/pip/data/management/service/LocationService.java

```diff
@@ -202,15 +227,71 @@ public byte[] downloadLocations() throws IOException {
      * This method will delete a location from the database.
      * @param locationId The ID of the location to delete.
      */
-    public void deleteLocation(Integer locationId) {
+    public LocationDeletion deleteLocation(Integer locationId, String requesterName)
+        throws JsonProcessingException {
+        LocationDeletion locationDeletion;
         Optional<Location> location = locationRepository.getLocationByLocationId(locationId);
 
         if (location.isPresent()) {
-            locationRepository.deleteById(locationId);
+            locationDeletion = checkActiveArtefactForLocation(location.get(), requesterName);
+            if (!locationDeletion.getIsExists()) {
+                locationDeletion = checkActiveSubscriptionForLocation(location.get(), requesterName);
+                if (!locationDeletion.getIsExists()) {
+                    locationRepository.deleteById(locationId);
+                    sendEmailToAllSystemAdmins(requesterName, ActionResult.SUCCEEDED,
+                        String.format("Location %s with Id %s has been deleted.",
+                                location.get().getName(), location.get().getLocationId()));
+                }
+            }
         } else {
             throw new LocationNotFoundException(
                 String.format("No location found with the id: %s", locationId));
         }
+
+        return locationDeletion;
+    }
+
+    private void sendEmailToAllSystemAdmins(String requesterName, ActionResult actionResult,
+                                            String additionalDetails) throws JsonProcessingException {
+        List<String> systemAdmins = accountManagementService.getAllAccounts("0",
+                                                                "1000", "PI_AAD");
+        publicationService.sendSystemAdminEmail(systemAdmins, requesterName, actionResult, additionalDetails);
+    }
+
+    private LocationDeletion checkActiveArtefactForLocation(Location location, String requesterName)
+        throws JsonProcessingException {
+        LocalDateTime searchDateTime = LocalDateTime.now();
+        LocationDeletion locationDeletion = new LocationDeletion();
+        List<Artefact> activeArtefacts =
+            artefactRepository.findActiveArtefactsForLocation(searchDateTime, location.getLocationId().toString());
+        if (!activeArtefacts.isEmpty()) {
+            buildErrorAndAdminAction(locationDeletion,true,
+                                     "There are active artefacts for the given location.");
+            sendEmailToAllSystemAdmins(requesterName, ActionResult.ATTEMPTED,
+                String.format("There are active artefacts for following location: %s", location.getName()));
+        }
+        return locationDeletion;
```

Comment:
Not sure if you need the class LocationDeletion at all. What you can do here is if activeArtefacts not empty, return Optional.of("There are active artefacts for the given location."). If it is empty, return Optional.empty().
You can then use the returned optional isPresent() to check whether there is any error message (i.e. active artefacts in the system)

Reply:
As discussed, I have refactored the code.